### PR TITLE
Modfication to the mt_shapes part to run properly on data

### PR DESCRIPTION
**Pull request by amagitte**
Hi Loic,

I have changed last week the last part where the mt_sahpes hits are filled in such a way that these plots are filled also in the case we are running on data.

Alessio

### CODE REVIEW

File: bin/hzz2l2v/runHZZ2l2vAnalysis.cc [hunk 1]
```diff
@@ -1397,27 +1397,32 @@ int main(int argc, char* argv[])
...
-               else		      btsfutil.modifyBTagsWithSF(hasCSVtag,sfl+sflunc,leff);
              }
-             else if(varyBtagDown) {
-               if(abs(flavId)==5)        btsfutil.modifyBTagsWithSF(hasCSVtag,sfb-sfbunc,beff);
-               else if(abs(flavId)==4)   btsfutil.modifyBTagsWithSF(hasCSVtag,sfb/5-2*sfbunc,beff);
-               else		      btsfutil.modifyBTagsWithSF(hasCSVtag,sfl-sflunc,leff);
-             }*/
 
-             if(hasCSVtag) nbtagsjets++;
+             if( hasCSVtag ) { nbtagsjets++; }
```
Comment (quertenmont) [inline bin/hzz2l2v/runHZZ2l2vAnalysis.cc]: was changed for no reason

File: bin/hzz2l2v/runHZZ2l2vAnalysis.cc [hunk 2]
```diff
@@ -1428,11 +1433,11 @@ int main(int argc, char* argv[])
          
            //re-assign the event category to take migrations into account
            TString evCat  = eventCategoryInst.GetCategory(tightVarJets,boson);
-           //TString evCat  = eventCategoryInst.GetCategory(selJets,boson);
            
            for(size_t ich=0; ich<chTags.size(); ich++){
 
-             TString tags_full=chTags[ich]+evCat;
+             TString tags_full=chTags[ich];
```
Comment (quertenmont) [inline bin/hzz2l2v/runHZZ2l2vAnalysis.cc]: why this change?
this is a mistake for me...

File: bin/hzz2l2v/runHZZ2l2vAnalysis.cc [hunk 3]
```diff
@@ -1397,27 +1397,32 @@ int main(int argc, char* argv[])
 
              //check b-tag
              if( pt < 30 || fabs(eta) > 2.5 ) continue;
-             if(!isMC) continue;
```
Comment (quertenmont) [inline bin/hzz2l2v/runHZZ2l2vAnalysis.cc]: I dont understand all these changes....
All the things that enters in this loop is only about systematic studies, so it applied only to MC

File: bin/hzz2l2v/runHZZ2l2vAnalysis.cc (hunk 2 above)
Comment (amagitte) [inline bin/hzz2l2v/runHZZ2l2vAnalysis.cc]: I forget to  remove it...
It necessary only to compute the alpha without the split in jet bin category.

**Comment by quertenmont** [comment]
I don't think these are good changes, please explain them

**Comment by quertenmont** [comment]
Alessio, you absolutely need to make sure of the code you commit
otherwise you introduce bugs in the common code.
So always do a vimdiff and make sure that all your changes have a good reason to be there

File: bin/hzz2l2v/runHZZ2l2vAnalysis.cc (hunk 3 above)
Comment (amagitte) [inline bin/hzz2l2v/runHZZ2l2vAnalysis.cc]: If I want to study the alfa using data I need that mt_sahpes_* are filled with them..
So 

if(!isMC) continue; 

prevent this so I decide to do:

if( isMC ){
...
}

I agree that an other solution may be to put this line

bool hasCSVtag ( jets[ijet].bDiscriminator("pfCombinedInclusiveSecondaryVertexV2BJetTags") > 0.423 );

above

if( !isMC) continue;

do you understand what I want to say?

Comment (quertenmont) [inline bin/hzz2l2v/runHZZ2l2vAnalysis.cc]: then do (!isMC && ivar>0)continue;

2015-08-24 17:14 GMT+02:00 Alessio <notifications@github.com>:

> In bin/hzz2l2v/runHZZ2l2vAnalysis.cc
> <https://github.com/quertenmont/2l2v_fwk/pull/46#discussion_r37762036>:
>
> > @@ -1397,27 +1397,32 @@ int main(int argc, char* argv[])
> >
> >               //check b-tag
> >               if( pt < 30 || fabs(eta) > 2.5 ) continue;
> > -             if(!isMC) continue;
>
> If I want to study the alfa using data I need that mt_sahpes_* are filled
> with them..
> So
>
> if(!isMC) continue;
>
> prevent this so I decide to do:
>
> if( isMC ){
> ...
> }
>
> I agree that an other solution may be to put this line
>
> bool hasCSVtag (
> jets[ijet].bDiscriminator("pfCombinedInclusiveSecondaryVertexV2BJetTags") >
> 0.423 );
>
> above
>
> if( !isMC) continue;
>
> do you understand what I want to say?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/quertenmont/2l2v_fwk/pull/46/files#r37762036>.
>



-- 
---------------------------------------------------------
Loic Quertenmont - European Organization for Nuclear Research (CERN) - CMS
experiment
Office CERN   : bldg.40, RB-08
Phone CERN    : +41 2276 71548
Mobile BE     : +32 474 831531
WWW           : http://www.fynu.ucl.ac.be/users/l.quertenmont/
